### PR TITLE
[Mono] Add SIMD intrinsics for Vector{64, 128} on Arm64 Part I

### PR DESCRIPTION
**Pull request by fanyang-mono**
This PR contributes to #64072

This PR intrinsify the following methods of Vector{64, 128}:

- Add
- AndNot
- BitwiseAnd
- BitwiseOr
- Divide
- Max
- Min
- Multiply
- Subtract
- Xor

### CODE REVIEW

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -685,6 +755,48 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 		int op = id == SN_GetLower ? OP_XLOWER : OP_XUPPER;
 		return emit_simd_ins_for_sig (cfg, klass, op, 0, arg0_type, fsig, args);
 	}
+	case SN_Max:
+	case SN_Min: {
+		int instc0 = -1;
+		if (arg0_type == MONO_TYPE_R4 || arg0_type == MONO_TYPE_R8) {
+			switch (id) {
+			case SN_Max:
+				instc0 = INTRINS_AARCH64_ADV_SIMD_FMAX;
```
Comment (vargaz) [inline src/mono/mono/mini/simd-intrinsics.c:764]: These should check for TARGET_ARM64.

File: src/mono/mono/mini/simd-intrinsics.c [hunk 2]
```diff
@@ -560,6 +560,8 @@ type_to_extract_op (MonoTypeEnum type)
 }
 
 static guint16 sri_vector_methods [] = {
+	SN_Add,
```
Comment (vargaz) [inline src/mono/mono/mini/simd-intrinsics.c:563]: Unrelated, but why aren't these implemented in managed code using
if (Arm64.IsSupported) ... etc.

File: src/mono/mono/mini/simd-intrinsics.c [hunk 3]
```diff
@@ -633,6 +643,66 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
...
+				break;
+			case SN_Subtract:
+				instc0 = OP_FSUB;
+				break;
+			case SN_Max:
+				instc0 = OP_FMAX;
+				break;
+			case SN_Min:
+				instc0 = OP_FMIN;
+				break;
```
Comment (simonrozsival) [inline src/mono/mono/mini/simd-intrinsics.c:666]: These two cases aren't reachable, are they? (same for the else branch)

Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c:666]: Good catch, forgot to delete this code path after implementing the other one.

File: src/mono/mono/mini/simd-intrinsics.c (hunk 2 above)
Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c:563]: Good question. This whole work was motivated by #49397. My understanding of the purpose of this work is to minimize the managed code that .NET customers need to author.

**Review by simonrozsival — APPROVED** [review]

**Review by vargaz — APPROVED** [review]

**Comment by fanyang-mono** [comment]
@imhameed could you take a look at the failure in llvmaot lane here, while I am on vacation? For some reason, the type does match for max intrinsics on x64 with sse. Looks like my fix for that wasn’t sufficient. Thanks in advance!

File: src/mono/mono/mini/mini-llvm.c [hunk 4]
```diff
@@ -7713,29 +7713,73 @@ process_bb (EmitContext *ctx, MonoBasicBlock *bb)
...
+						iid = is_r4 ? INTRINS_SSE_MAXPS : INTRINS_SSE_MAXPD;
+				} else {
+					if (elems == 1)
+						iid = is_r4 ? INTRINS_SSE_MINSS : INTRINS_SSE_MINSD;
+					else
+						iid = is_r4 ? INTRINS_SSE_MINPS : INTRINS_SSE_MINPD;
+				}
+				result = call_intrins (ctx, iid, args, dname);
+#elif defined(TARGET_ARM64)
+				int instc0_arm64 = ins->inst_c0 == OP_FMAX ? INTRINS_AARCH64_ADV_SIMD_FMAX : INTRINS_AARCH64_ADV_SIMD_FMIN;
```
Comment (vargaz) [inline src/mono/mono/mini/mini-llvm.c:7735]: This could just set 'iid'.

File: src/mono/mono/mini/mini-llvm.c [hunk 5]
```diff
@@ -7713,29 +7713,73 @@ process_bb (EmitContext *ctx, MonoBasicBlock *bb)
...
+						instc0_arm64 = is_unsigned ? INTRINS_AARCH64_ADV_SIMD_UMAX : INTRINS_AARCH64_ADV_SIMD_SMAX;
+						break;
+					case OP_IMIN:
+						instc0_arm64 = is_unsigned ? INTRINS_AARCH64_ADV_SIMD_UMIN : INTRINS_AARCH64_ADV_SIMD_SMIN;
+						break;
+					default:
+						g_assert_not_reached ();
+					}
+					IntrinsicId iid = (IntrinsicId) instc0_arm64;
+					LLVMValueRef args [] = { l, r };
```
Comment (vargaz) [inline src/mono/mono/mini/mini-llvm.c:7775]: Ditto.

**Review by vargaz — APPROVED** [review]

**Review by ihameed — APPROVED** [review]

File: src/mono/mono/mini/mini-llvm.c (hunk 4 above)
Comment (fanyang-mono) [reply src/mono/mono/mini/mini-llvm.c:7735]: Will fix this in a follow-up PR. I'd like to get this merged before the Preview2 snap.

Comment (fanyang-mono) [reply src/mono/mono/mini/mini-llvm.c:7735]: Fixed via #65445

File: src/mono/mono/mini/mini-llvm.c (hunk 5 above)
Comment (fanyang-mono) [reply src/mono/mono/mini/mini-llvm.c:7775]: Fixed via #65445

**Pull request merged**